core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala

```diff
@@ -561,6 +561,108 @@ class RemoteIndexCacheTest {
+      s"Offset index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTimeIndexFileName(missingMetadata)).isPresent,
+      s"Time index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTransactionIndexFileName(missingMetadata)).isPresent,
+      s"Txn index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteDeletedSuffixIndexFileName(missingMetadata)).isPresent,
+      s"Index file marked for deletion for evicted entry should not be present on disk at ${cache.cacheDir()}")
+
+    // Reduce cache capacity to only store 1 entries
+    cache.resizeCacheSize(1 * estimateEntryBytesSize)
+    assertCacheSize(1)
+
+    val nextMissingMetadata = metadataList(1)
+    val nextMissingEntry = entry1
+    // wait until evicted entry is marked for deletion
+    TestUtils.waitUntilTrue(() => nextMissingEntry.isMarkedForCleanup,
+      "Failed to mark evicted cache entry for cleanup after resizing cache.")
+    TestUtils.waitUntilTrue(() => nextMissingEntry.isCleanStarted,
+      "Failed to cleanup evicted cache entry after resizing cache.")
+    // verify no index files for `nextMissingEntry` on remote cache dir
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteOffsetIndexFileName(nextMissingMetadata)).isPresent,
+      s"Offset index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTimeIndexFileName(nextMissingMetadata)).isPresent,
+      s"Time index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteTransactionIndexFileName(nextMissingMetadata)).isPresent,
+      s"Txn index file for evicted entry should not be present on disk at ${cache.cacheDir()}")
+    TestUtils.waitUntilTrue(() => !getIndexFileFromRemoteCacheDir(remoteDeletedSuffixIndexFileName(nextMissingMetadata)).isPresent,
+      s"Index file marked for deletion for evicted entry should not be present on disk at ${cache.cacheDir()}")
+
+    // verify the existing entry is of `cache.getIndexEntry(metadataList(2))`
+    assertTrue(getIndexFileFromRemoteCacheDir(remoteOffsetIndexFileName(metadataList(2))).isPresent)
+    assertTrue(getIndexFileFromRemoteCacheDir(remoteTimeIndexFileName(metadataList(2))).isPresent)
+    assertTrue(getIndexFileFromRemoteCacheDir(remoteTransactionIndexFileName(metadataList(2))).isPresent)
+    assertTrue(!getIndexFileFromRemoteCacheDir(remoteDeletedSuffixIndexFileName(metadataList(2))).isPresent)
+    assertTrue(cache.internalCache().estimatedSize() == 1)
+  }
```

Comment:
Could we have a final test case that if we resize to a bigger size, the existing cache (i.e. `metadataList(2)`) will still be kept in cache?

Reply:
added.